felix/bpf-gpl/tc.c

```diff
@@ -559,6 +559,34 @@ static CALI_BPF_INLINE void calico_tc_process_ct_lookup(struct cali_tc_ctx *ctx)
 		goto do_policy;
 	}
 
+	/* If the dest route is a blackhole route and packet is from an external source,
+	 * drop the packet, if service loop prevention is configured to either DROP or REJECT.
+	 */
+	if (cali_rt_is_blackhole(dest_rt) && !(ctx->state->flags & CALI_ST_SKIP_FIB)) {
+		if (CALI_F_FROM_HEP && !(cali_rt_lookup(&ctx->state->ip_src))) {
+			struct cali_rt *src_rt = cali_rt_lookup(&ctx->state->ip_src);
```

Comment:
Why do you test above `if (!(cali_rt_lookup(&ctx->state->ip_src))) {` and then you do the same lookup here again :thinking: